### ISSUE TITLE
add build output option

### ISSUE THREAD

**Issue by mathisonian** [idyll-lang/idyll#16]
Right now Idyll just expects you to use the `./build` directory when you build a version of a project. This should be an option instead

**Comment by rreusser** [comment]
TBH, it caught me off guard when I realized `--build` output to stdout *and* wrote files at the same time. TeX takes a different approach and doesn't output anything to stdout, which IMO turns out to be surprisingly inconvenient. In particular it makes it impossible to pipe anything at all and instead forces a mess of tmp files/folders. Though I understand why it has to be that way. As I mentioned briefly, I've been using hyperstream tools a lot lately ([simple-html-index](https://github.com/mattdesl/simple-html-index), [html-inject-index](https://github.com/rreusser/html-inject-script), [metadataify](https://github.com/rreusser/metadataify), [github-cornerify](https://github.com/rreusser/github-cornerify), others?) to focus on generating files programatically instead of starting from a template and filling in the blanks.

So this doesn't totally fit with this issue, but backing up just a small step, I was wondering if there's any conceivable way to rework the build step to send all results to stdout. Which really means just breaking up `--build`. The first option is `--bundle` and `--build-html` which is a bit clunky. So maybe there should be three steps in order to avoid potential inconsistency: 1. `--build` constructs some intermediate representation of the output, 2. process the result of step 1 into a js bundle, 3. process the result of step 1 into html. That might be way worse though. Or maybe the html build option just uses hyperstream to inject the entry point into some selector?

It may be best to disregard all of this. I'm doing my best to suggest alternatives since the html output felt like a surprising side effect of building, but I'm coming up a bit short. 😞

**Comment by mathisonian** [comment]
I agree the current build setup is totally weird and I'd like to fix it. To me the model of not outputting anything to stdout seems simpler overall. If scripting the html is the main concern it that could be done in its own step to generate the html template that Idyll uses as input:

```
$ metadataify < _template.html > _index.html
$ idyll --build 
```

I don't think its as flexible as what you're proposing but a bit easier to work with and could still be scriptable if you wanted to configure the build command yourself in javascript.

**Comment by rreusser** [comment]
That seems perfectly reasonable. Is there much of a reason to pipe the bundle beyond just sending it to a file?

**Comment by mathisonian** [comment]
I think the only reason it started off as a pipe was to quickly use `uglify`, but that could be done programmatically before sending it to a file.

**Comment by mathisonian** [comment]
fixed in #18
